src/event_handler.rs

```diff
@@ -504,63 +511,148 @@ pub async fn handle_event_logs(
 // Function to regularly check a transaction for block confirmation and resend it if not included
 async fn resend_pending_transaction(
     app_state: Data<AppState>,
-    txn: FunctionCall<Arc<HttpSignerProvider>, HttpSignerProvider, ()>,
-    mut pending_txn_hash: H256,
-    nonce: U256,
-    mut gas_limit: U256,
-    mut gas_price: U256,
-    resend_deadline: u128,
+    pending_txns: Arc<Mutex<VecDeque<PendingTxnData>>>,
 ) {
-    // Calculating resend retries number based on the overall deadline and interval in which to resend pending/dropped txns
-    let mut resend_retries = resend_deadline / (RESEND_TXN_INTERVAL as u128);
-    while resend_retries > 0 {
-        sleep(Duration::from_secs(RESEND_TXN_INTERVAL)).await;
-
-        // Get the transaction receipt for the pending transaction to check if it's still pending or been dropped
-        let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
-        let Ok(txn_receipt) = http_rpc_client
-            .get_transaction_receipt(pending_txn_hash)
-            .await
-        else {
-            resend_retries -= 1;
+    loop {
+        if pending_txns.lock().unwrap().is_empty() {
+            sleep(Duration::from_millis(200)).await;
             continue;
-        };
+        }
 
-        if txn_receipt.is_some() {
-            break;
+        let mut pending_txn_data = pending_txns.lock().unwrap().pop_front().unwrap();
```

Comment:
remove the if condition above, and handle the pop_front return value (i.e. None if empty) for sleep and retry.